### DISCUSSION TITLE
After the recent Unity mess, could Adobe AIR be considered an interesting alternative?

### DISCUSSION THREAD

**Discussion by Michoko92** [airsdk/Adobe-Runtime-Support#2828] [General]
Hi,

I suppose you heard of the recent Unity pricing policy change, that will make it much less interesting for Indie Developers (if not, you can have a look here, for example : https://www.gamedeveloper.com/blogs/unity-s-just-not-into-you-indie-developer)

Many small developers are now looking for alternatives, Godot seemingly being the new cool kid in town here. But I'm not sure that it has even a solid native libraries offer as we have with the Distriqt ANEs. I still believe the AIR platform has very strong qualities, and it's really a shame it is not considered as a valid alternative. I know the ghost of Flash is still floating around, but I think Harman did a surprisingly amazing job at maintaining the platform and listening to the community. It might be time to promote the AIR SDK more, and present it as an interesting alternative for many cross-platform projects.

Harman, are you considering doing more promotional work in the upcoming weeks/months?

**Comment by marchbold** [comment]
Totally agree, I think it's getting to be a good time to start promoting AIR! What that looks like I'm not really sure, but it's definitely back to a point where I am confident to develop apps in the platform again.

**Reply by teotigraphix** [reply]
> but it's definitely back to a point where I am confident to develop apps in the platform again.

That is interesting to say.

**Reply by marchbold** [reply]
Is it? Harman have been doing a great job at maintaining and feature adding the platform and seem to have aspirations to take the platform further. So yes gives me confidence.

**Reply by teotigraphix** [reply]
> yes gives me confidence.

.. The non interesting way. ;-) (was the **again** reference)

**Reply by marchbold** [reply]
Ah I see. After years of fighting with / pestering Adobe for bug fixes and making workarounds to get simple native functionality to work in AIR, definitely had lost confidence in the platform, so the *again* is very relevant to me :)

**Comment by Devin-Licastro** [comment]
Air's time to shine!

**Comment by Michoko92** [comment]
I actually posted a topic here: https://forum.starling-framework.org/d/23858-air-as-a-valid-game-engine-alternative as an attempt to list all AIR features that can make it a valid  solution for game development. Please feel free to share your experience and technical knowledge, it will be very useful to build an updated list of viable solutions. Cheers!

**Comment by shaucky** [comment]
Although this is not good, but forgive me for throwing a damp over. AIR lacks an IDE for game development like Unity Editor or Unreal Editor, especially for 3D development, which seems even more troublesome… 
 
Is it possible for the development community to work together to build an IDE that can integrate various popular AIR frameworks (such as Starling and Away3D) and ActionScript compilers? I think some of the designs of Unity Editor are quite interesting, such as customizing C# to adjust the IDE menu. But this may be a huge project,  just think about it…
 
Putting aside these relatively weak aspects, i'm still full of emotions towards AIR. If AIR can stand out in such a situation, it is truly gratifying.

**Reply by FischMich** [reply]
Revive FLARE 3D! 
https://www.flare3d.com/

**Reply by FischMich** [reply]
Unfortunately also discontinued: https://quentinlengele.com/index.php/2011/06/08/worldbuilder/

**Reply by Dextercitox** [reply]
I dont think an IDE is needed. I've been working with intelliJ last 4 years using Starling and everything goes like a charm, old Flash IDE was useful when I started programming but when you go forward, it's not needed at all. :)

**Reply by shaucky** [reply]
Yeah, for experienced developers, IDEs are relatively less useful. But for beginners who are not familiar with frameworks, IDEs are beneficial for familiarizing themselves with development, i think so. And also the Flash Pro is outdated…

**Comment by pnico** [comment]
It's funny because both Unity and Unreal (and possibly Godot, idk) were 3D-first engines and then developed more support for 2D later because lots of developers just wanted to make 2D games and didn't care at all about the fancy 3D features, they wanted the other benefits of those engines like performance, asset management, console support, etc. and were happy to pay a reasonable price for them so there was a market for that. There are still, and will probably always be, lots of developers who don't care about 3D at all. AIR still has features for 2D vector animation (OK, so performance isn't the greatest but it looks good) that these engines don't match except with some addon like Rive. I'm not saying you can't or shouldn't try to do 3D things with AIR, but it's very not-3D-first.

That was a rambling paragraph just to preface a suggestion that I know will never happen: Provide console support for AIR on the level that Unity does, grab all those disillusioned 2D game developers who just want the option of releasing their indie game on Nintendo Switch someday 😂
